cocos/animation/marionette/animation-graph-context.ts

```diff
@@ -106,11 +129,46 @@ export class AnimationGraphBindingContext {
         return this._varRegistry[id];
     }
 
+    /**
+     * Pushes the `additive` flag. A later `_popAdditiveFlag` is required to pop the change.
+     * @internal
+     */
+    public _pushAdditiveFlag (additive: boolean) {
+        this._additiveFlagStack.push(additive);
```

Comment:
Performance concerns:

The `_pushAdditveFlag`/`_popAdditiveFlag` will called in bind stage(only one times in intialization, or caused by `overrideClips`). Each layer will call once. In future there will be a pose node `Add`, this pose node will call this pair of methods too.